core/src/main/java/org/jboss/pnc/rex/core/RemoteEntityClient.java

```diff
@@ -72,23 +74,46 @@ public class RemoteEntityClient {
 
     private final String baseUrl;
 
-    @Inject
-    Tokens serviceTokens;
+    private final Tokens serviceTokens;
 
     public RemoteEntityClient(GenericVertxHttpClient client,
                               @WithTransactions TaskController controller,
                               TaskRegistry taskRegistry,
                               ObjectMapper mapper,
                               @ConfigProperty(name = "scheduler.baseUrl",
-                                      defaultValue = "http://localhost:8080") String baseUrl) {
+                                      defaultValue = "http://localhost:8080") String baseUrl,
+                              Tokens serviceTokens) {
         this.controller = controller;
         this.taskRegistry = taskRegistry;
         this.client = client;
         this.mapper = mapper;
         this.baseUrl = baseUrl;
+        this.serviceTokens = serviceTokens;
     }
 
     public void stopJob(Task task) {
+        if (task.getConfiguration() != null && task.getConfiguration().getMdcHeaderKeys() != null) {
+            var keys = task.getConfiguration().getMdcHeaderKeys();
+            var headers = task.getRemoteCancel().getHeaders().stream().collect(Collectors.toMap(Header::getName, Header::getValue));
+            
+            wrapWithMDC(keys, headers, () -> stopJobInternal(task));
+        } else {
+            stopJobInternal(task);
+        }
+    }
+
+    public void startJob(Task task) {
+        if (task.getConfiguration() != null && task.getConfiguration().getMdcHeaderKeys() != null) {
+            var keys = task.getConfiguration().getMdcHeaderKeys();
```

Comment:
As far as I know we have an established list of MDC headers that we use. Is there a reason to make MDC headers customizable?

Reply:
It fits Rex more as it is still a generic task scheduler. I'd like to avoid it being baked for PNC.